### PR TITLE
LPS-11010

### PR DESCRIPTION
**Pull request by Samireddebbarhi**

### CODE REVIEW

**Comment by pedroamarall** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/paths/ConfigurationPanel.path [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+<tbody>
+<tr>
+	<td>CONFIGURATION_H2</td>
+	<td>//*[@id="nav_control_panel.configuration"][contains(.,'Configuration')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>BUTTON_MENU</td>
+	<td>//*[@id="controlMenu"]/div[4]/ul/li[7]/button</td>
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/paths/ConfigurationPanel.path:20]: change it to a relative xpath

File: portal-web/test/functional/com/liferay/portalweb/paths/ConfigurationPanel.path [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>BUTTON_CONTROL</td>
+	<td>//ul[contains(@class,'nav nav-tab')]/li/button/span[contains(.,'Control Panel')]</td>
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/paths/ConfigurationPanel.path:25]: change it to a relative xpath

File: portal-web/test/functional/com/liferay/portalweb/paths/CustomSimulation.path [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+<tbody>
+<tr>
+	<td>APPLY_CUSTOM</td>
+	<td>//button[contains(.,'Apply Custom Size')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>APPLIED_CUSTOM</td>
+	<td>//div[contains(@class,'autofit-col autofit-col-expand')]//div[contains(.,'Custom Size 1920x1080 Applied')]</td>
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/paths/CustomSimulation.path:20]: do you really need to catch the classes with this order? what happens if the programmer changes the order?
about the div, do you really need to define the div? what happens if the programmer changes it to a "p" ?

File: portal-web/test/functional/com/liferay/portalweb/tests/Formtest.testcase [hunk 4]
```diff
@@ -0,0 +1,136 @@
+@component-name = "portal-web-forms-and-data-lists"
+definition {
+
+	//property osgi.module.configuration.file.names = "com.liferay.frontend.js.loader.modules.extender.internal.configuration.Details.config";
+	//property osgi.module.configurations = "waitTimeout=\"15\"";
+	//property portal.release = "true";
+	//property portal.upstream = "true";
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/Formtest.testcase:7]: remove the comment lines, that is not necessary.

File: portal-web/test/functional/com/liferay/portalweb/tests/Formtest.testcase [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+			Form.editFieldLabelMetal(fieldValue = "Full Name");
+
+			Form.save();
+
+			Form.gotoAddField(fieldType = "Numeric");
+
+			Form.editFieldLabelMetal(fieldValue = "Age");
+
+			Form.save();
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/Formtest.testcase:56]: On the setup you are creating and editing, do you really need it?
can you change it to a JSON call?

File: portal-web/test/functional/com/liferay/portalweb/tests/Formtest.testcase [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+			ProductMenu.gotoPortlet(
+				category = "Content & Data",
+				portlet = "Forms");
+
+			FormsAdminNavigator.gotoViewEntries();
+
+			LexiconEntry.gotoEntryMenuItem(
+				menuItem = "View",
+				rowEntry = "Samireddebbarhi");
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/Formtest.testcase:132]: Then should validate / assert the things, but here you are changing something, so, it should be here?

File: portal-web/test/functional/com/liferay/portalweb/tests/MyBlogTests.testcase [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+				category = "Content & Data",
+				portlet = "Blogs");
+		}
+
+		task ("When the user adds a Blog Entry") {
+			JSONBlog.addEntry(
+				entryContent = "Blogs",
+				entryTitle = "Test");
+
+			Refresh();
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/MyBlogTests.testcase:36]: the When is about what do you want to test, so, it should not be performed by a JSON call...

File: portal-web/test/functional/com/liferay/portalweb/tests/MyBlogTests.testcase [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+		task ("Given the Blogs Portlet is added to the page And Blog Entry exists ") {
+			MyBlogs.Addblog();
+		}
+
+		task ("When the Blogs Portlet is deleted from the page") {
+			JSONBlog.moveEntryToTrash(
+				entryTitle = "Test",
+				groupName = "Guest");
+
+			Refresh();
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/MyBlogTests.testcase:59]: the When is about what do you want to test, so, it should not be performed by a JSON call...

File: portal-web/test/functional/com/liferay/portalweb/tests/MyBlogTests.testcase [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+				rowEntry = "Test");
+
+			BlogsEntry.editTitle(
+				entryTitle = "Test",
+				entryTitleEdit = "New title");
+		}
+
+		task ("Then the Blog Entry should be edited and published") {
+			Button.clickPublish();
+		}
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/MyBlogTests.testcase:86]: are you validating that it really was created/edited as you defined on the test?

File: portal-web/test/functional/com/liferay/portalweb/tests/MyBlogTests.testcase [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+				entryTitle = "Test");
+		}
+
+		task ("Then Blog Entry should display ") {
+			AssertElementPresent(
+				key_entryTitle = "Test",
+				key_tagName = "Test",
+				locator1 = "BlogsEntry#TAG");
+		}
+	}
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/MyBlogTests.testcase:108]: perfect test!

but please, review the name of your test to tell what you will test....

"Can search blogs", with success? failure? will detect something? will see if some wrong result appears?

**Review by manoelcyreno — CHANGES_REQUESTED** [review]

**Comment by pedroamarall** [comment]
Follow in here [LPS-10011](https://github.com/pedroamarall/liferay-portal/pull/60#top)

**Pull request closed without merging**